### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by quver**
NSMakeRect -> NSRect
CGPointMake -> CGPoint
Unnecesery self removed
some complex if's change to if where
.gitingore added

### CODE REVIEW

**Comment by michaelvillar** [comment]
Can you make sure to use the same indentation? It's difficult to see the diff.
I use
- tab width: 2 spaces
- indent width: 2 spaces

**Comment by quver** [comment]
Done :)

File: Timer/AppDelegate.swift [hunk 1]
```diff
@@ -25,28 +25,27 @@ class AppDelegate: NSObject, NSApplicationDelegate, NSUserNotificationCenterDele
...
-      if controller != nil {
-        let index = controllers.indexOf(controller!)
-        if index != nil {
-          controllers.removeAtIndex(index!)
-        }
-      }
+
+    guard  let window = notification.object as? NSWindow,
+      let controller = controllerForWindow(window),
+      let index = controllers.indexOf(controller) else {
```
Comment (michaelvillar) [inline Timer/AppDelegate.swift:42]: weird indentation here?

File: Timer/AppDelegate.swift [hunk 2]
```diff
@@ -25,28 +25,27 @@ class AppDelegate: NSObject, NSApplicationDelegate, NSUserNotificationCenterDele
...
-    if let window = notification.object as? NSWindow {
-      let controller = self.controllerForWindow(window)
-      if controller != nil {
-        let index = controllers.indexOf(controller!)
-        if index != nil {
-          controllers.removeAtIndex(index!)
-        }
-      }
+
+    guard  let window = notification.object as? NSWindow,
```
Comment (michaelvillar) [inline Timer/AppDelegate.swift:40]: extra space between guard and let?

File: Timer/MVClockView.swift [hunk 3]
```diff
@@ -15,277 +15,277 @@ class MVClockView: NSControl {
...
-    ])
+      ])
     minutesLabelSuffixWidth = minutesLabelSize.width
-    
+
     let minutesLabelSecondsSuffix = "\""
     let minutesLabelSecondsSize = minutesLabelSecondsSuffix.sizeWithAttributes([
       NSFontAttributeName: minutesLabel.font!
-    ])
+      ])
```
Comment (michaelvillar) [inline Timer/MVClockView.swift:91]: extra space

File: Timer/MVClockView.swift [hunk 4]
```diff
@@ -15,277 +15,277 @@ class MVClockView: NSControl {
...
     minutesLabel.textColor = NSColor(SRGBRed: 0.2353, green: 0.2549, blue: 0.2706, alpha: 1.0)
-    self.addSubview(minutesLabel)
-    
+    addSubview(minutesLabel)
+
     let minutesLabelSuffix = "'"
     let minutesLabelSize = minutesLabelSuffix.sizeWithAttributes([
       NSFontAttributeName: minutesLabel.font!
-    ])
+      ])
```
Comment (michaelvillar) [inline Timer/MVClockView.swift:85]: extra space

File: Timer/MVClockView.swift [hunk 5]
```diff
@@ -15,277 +15,277 @@ class MVClockView: NSControl {
...
     secondsLabel.textColor = NSColor(SRGBRed: 0.6353, green: 0.6667, blue: 0.6863, alpha: 1.0)
-    self.addSubview(secondsLabel)
-    
+    addSubview(secondsLabel)
+
     let secondsLabelSuffix = "'"
     let secondsLabelSize = secondsLabelSuffix.sizeWithAttributes([
       NSFontAttributeName: secondsLabel.font!
-    ])
+      ])
```
Comment (michaelvillar) [inline Timer/MVClockView.swift:103]: extra space

**Comment by michaelvillar** [comment]
Feels a lot like you have different conventions than mine. I actually really like `self.method`, it's clear that you call a instance method and not a global function.

Not sure if this PR is required.

Sorry.

File: Timer/AppDelegate.swift (hunk 1 above)
Comment (quver) [inline Timer/AppDelegate.swift:42]: This is default Xcode indentation in this case

**Comment by quver** [comment]
Many guidelines like raywenderlich.com suggest you shouldn't use self

>  For conciseness, avoid using self since Swift does not require it to access an object's properties or invoke its methods.
> 
> Use self when required to differentiate between property names and arguments in initializers, and when referencing properties in closure expressions (as required by the compiler)

**Pull request closed without merging**